### DISCUSSION TITLE
Call to undefined method when using nested relationships on aggregate function

### DISCUSSION THREAD

**Discussion by adesege** [laravel/framework#38608] [Q&A]
- Laravel Version: 8.57.0
- PHP Version: 8.0
- Database Driver & Version: Postgres 13

### Description:
`Call to undefined method` error is thrown when using nested relationship on aggregate functions like `withAvg`. I was able to debug this and found out that `getRelationWithoutConstraints` is directly calling the relation parameter as a function.

In my use case, I'm trying to get the average of book reviews from a shelf model.

```php
Shelf::withAvg('books.reviews', 'rating');
```

Getting the reviews directly from the book model works fine, however.

Related to #38049 and #38079 

### Steps To Reproduce:

1. Create 3 models with the following relationships; A -> B -> C.
1. Then get the average of `C` through model A; `A::withAvg('b.c', 'd')`; // This will throw an error
1. If you do `B::withAvg('c', 'd')` instead, it will not throw an error.

**Comment by staudenmeir** [comment]
`withAvg()` etc. don't (and basically can't) work with nested relationships. You need to build a direct relationship between `Shelf` and `Review`. How are `books` and `reviews` defined?

**Reply by adesege** [reply]
> `withAvg()` etc. don't (and basically can't) work with nested relationships. You need to build a direct relationship between `Shelf` and `Review`. How are `books` and `reviews` defined?

@staudenmeir Thanks for the clarification. Books hasMany Reviews while Shelf belongs to many Books. I think the clarification that `withAvg` can't work with nested relationships should be included in the documentation. Spent some days scratching my head on this.

**Comment by staudenmeir** [comment]
The aggregate functions could also throw an exception when being called with a nested relationship.

You can define a direct relationship by "skipping" the `books` table:

```php
class Shelf extends Model
{
    public function reviews()
    {
        return $this->belongsToMany(
            Review::class,
            'book_shelf',
            'shelf_id', 'book_id', null, 'book_id'
        );
    }
}

Shelf::withAvg('reviews', 'rating');
```
